### PR TITLE
Add documentation for using Private Link and External DNS

### PR DESCRIPTION
**Pull request by jnpacker**
Signed-off-by: Joshua Packer <jpacker@redhat.com>

# Description of the change(s):
* Documentation for Private Link (Private Clusters)
* Documentation for External DNS

## Why do we need this PR:
* Service Delivery
* Hypershift supported configurations

## Issue reference: 
* https://issues.redhat.com/browse/ACM-1558
* https://issues.redhat.com/browse/ACM-1474

## Test API/Unit - Success
```script
NA
```

### CODE REVIEW

File: docs/provision_hypershift_clusters_by_mce.md [hunk 1]
```diff
@@ -37,129 +37,223 @@ We call the cluster with the HyperShift operator installed as the hosting servic
...
+    * IAM policy in AWS
+    * Hypershift operator IAM user
+    * Attach the policy to the user
+    * Create an IAM access key for the user, use the access key and secret in the next section
+
+    The secret must contain 3 fields:
+
+    - `aws-access-key-id`: AWS credential access key id
+    - `aws-secret-access-key`: AWS credential access key secret
+    - `region`: Region for use with Private Linke
```
Comment (rokej) [inline docs/provision_hypershift_clusters_by_mce.md]: Typo `Linke`

**Review by rokej — COMMENTED** [review]
Thanks! I found just one type but the rest looks good to me.

File: docs/provision_hypershift_clusters_by_mce.md [hunk 2]
```diff
@@ -37,129 +37,223 @@ We call the cluster with the HyperShift operator installed as the hosting servic
...
+    spec:
+      hostedClusterSpec:
+        platform:
+          type: AWS
+          aws:
+            endpointAccess: Private
+    ```
+
+    ### External DNS
+    If you plan to use service-level DNS for Control Plane Service, create an external DNS credential secret for the HyperShift operator, and name it `hypershift-operator-external-dns-credentials`. It should reside in managed cluster namespace (i.e., the namespace of the managed cluster that will be used as the hosting service cluster). If you used `local-cluster`, then create the secret in the `local-cluster` namespace
```
Comment (philipwu08) [inline docs/provision_hypershift_clusters_by_mce.md]: "in managed cluster namespace" => "in the managed cluster namespace". Is also missing for the private link and default s3 creds section.

Do we want a period at the end?

File: docs/provision_hypershift_clusters_by_mce.md (hunk 1 above)
Comment (philipwu08) [inline docs/provision_hypershift_clusters_by_mce.md]: "Linke" => "Link"

File: docs/provision_hypershift_clusters_by_mce.md [hunk 3]
```diff
@@ -37,129 +37,223 @@ We call the cluster with the HyperShift operator installed as the hosting servic
...
+    If you plan to provision hosted clusters on the AWS platform with Private Link, create an AWS credential secret for the HyperShift operator, and name it `hypershift-operator-private-link-credentials`. It should reside in managed cluster namespace (i.e., the namespace of the managed cluster that will be used as the hosting service cluster). If you used `local-cluster`, then create the secret in the `local-cluster` namespace
+
+    Follow the instructions here (steps 1-5): [Deploy AWS private clusters](https://hypershift-docs.netlify.app/how-to/aws/deploy-aws-private-clusters/)
+
+    Instructions result in the following AWS resources:
+
+    * IAM policy in AWS
+    * Hypershift operator IAM user
+    * Attach the policy to the user
+    * Create an IAM access key for the user, use the access key and secret in the next section
```
Comment (philipwu08) [inline docs/provision_hypershift_clusters_by_mce.md:70]: ", use" , maybe add "and" or put the "user ...." in parenthesis?

File: docs/provision_hypershift_clusters_by_mce.md (hunk 1 above)
Comment (jnpacker) [reply docs/provision_hypershift_clusters_by_mce.md]: Corrected!

File: docs/provision_hypershift_clusters_by_mce.md (hunk 2 above)
Comment (jnpacker) [reply docs/provision_hypershift_clusters_by_mce.md]: Added `the` to all three. The period is after the (i.e,...) section

File: docs/provision_hypershift_clusters_by_mce.md (hunk 3 above)
Comment (jnpacker) [reply docs/provision_hypershift_clusters_by_mce.md:70]: Reworded

**Review by rokej — APPROVED** [review]

**Pull request merged**